systemd-swap

```diff
@@ -70,7 +70,19 @@ get_fs_type(){
   elif [ -d $(dirname "$1") ]; then path=$(dirname "$1");
   else ERRO "swapfc_path is invalid"
   fi
-  df "$path" --output=fstype | tail -n 1
+
+  fstype=df "$path" --output=fstype | tail -n 1
+
+  if [ "$fstype" == "-" ]; then
+    btrfs subvolume show "$path" &>/dev/null
+    if [ "$?" -eq 0 ]; then
```

Comment:
You don't need to explicitly check for the exit code in another test, you can write that as:

```shell
if btrfs subvolume show "$path" &>/dev/null; then
```

Reply:
Stupid of me not to notice that, thanks